lib/posix.lua.in

```diff
@@ -519,5 +519,27 @@ if _DEBUG ~= false then
   end
 end
 
+local function prepare_args (pat_mark)
+  local pat, mark = "*", 0
+  if type (pat_mark) == "table" then
+    if type (pat_mark.pattern) == "string" then
+      pat = pat_mark.pattern
+    end
+    if pat_mark.MARK then
+      mark = M.GLOB_MARK
+    end
+  elseif type (pat_mark) == "string" then
+    pat = pat_mark
+  end
+
+  return pat, mark
+end
+
+local cglob = M.glob
+
+M.glob = function (...)
+  local pat, mark = prepare_args(...)
+  return cglob(pat, mark)
```

Comment:
Here I would save a stack frame and collect and parse the Lua API args inside this one function:

```lua
--- Find all files in this directory matching a shell pattern.
-- Brief description of MARK effects on the result here...
-- @function glob
-- @param[opt={}] string|table args either a glob pattern string,
--    or a table with optional keys `pattern` and `MARK`.
-- @return etc. etc.

M.glob = function (args)
  -- support previous `glob ".*"` style calls for backwards compatibility
  if type(args) == "string" then
    args = {pattern=args}
  elseif type(args) ~= "table" then
    args = {}
  end
  local flags = 0
  if args.MARK then
    flags = posix.glob.GLOB_MARK
  end
  -- in future we can add support for any further glob flags here
  return posix.glob.glob (args.pattern, flags)
end
```